packages/app/src/components/Randomizer/Randomizer.test.js

```diff
@@ -43,45 +44,45 @@ test('restaurant url is not rendered initially (no restaurant is shown)', async
 test('restaurant url is rendered after a restaurant is drawn', async () => {
   const { queryByTestId, getByTestId } = await actRender(<Randomizer />)
 
-  fireEvent.click(queryByTestId('randomizer-randomizeButton'))
-  await waitForDomChange()
-
-  const url = getByTestId('randomizer-restaurantUrl')
-  expect(url).toBeInTheDocument()
+  await act(async () => {
+    fireEvent.click(queryByTestId('randomizer-randomizeButton'))
+    await waitForDomChange()
+    const url = getByTestId('randomizer-restaurantUrl')
+    expect(url).toBeInTheDocument()
+  })
 })
 
 test('pressing the button calls the restaurant service', async () => {
   const { queryByTestId } = await actRender(<Randomizer />)
-  const buttonElement = queryByTestId('randomizer-randomizeButton')
-
-  fireEvent.click(buttonElement)
-  await waitForDomChange()
-
-  expect(restaurantService.getRandom).toBeCalled()
+  await act(async () => {
+    fireEvent.click(queryByTestId('randomizer-randomizeButton'))
+    expect(restaurantService.getAllMatches).toBeCalled()
+  })
 })
 
 test('pressing the button changes the text', async () => {
   const { queryByTestId } = await actRender(<Randomizer />)
 
   const restaurantNameElement = queryByTestId('randomizer-resultLabel')
   const originalText = restaurantNameElement.textContent
-
-  fireEvent.click(queryByTestId('randomizer-randomizeButton'))
-  await waitForDomChange()
-
-  expect(restaurantNameElement.textContent).not.toBe(originalText)
+  await act(async () => {
+    fireEvent.click(queryByTestId('randomizer-randomizeButton'))
+    await waitForDomChange()
+    expect(restaurantNameElement.textContent).not.toBe(originalText)
+  })
 })
 
 test('user is not redirected to an external website if not confirmed', async () => {
   const { queryByTestId } = await actRender(<Randomizer />)
 
   // Generate a restaurant suggestion
-  fireEvent.click(queryByTestId('randomizer-randomizeButton'))
-  await waitForDomChange()
-
-  const url = queryByTestId('randomizer-restaurantUrl')
-  window.confirm = jest.fn(() => false)
-  fireEvent.click(url)
-
-  expect(url).toBeInTheDocument()
+  await act(async () => {
+    fireEvent.click(queryByTestId('randomizer-randomizeButton'))
+    await waitForDomChange()
+    const url = queryByTestId('randomizer-restaurantUrl')
+    window.confirm = jest.fn(() => false)
+    fireEvent.click(url)
+    expect(url).toBeInTheDocument()
+  })
```

Comment:
```suggestion
  window.confirm = jest.fn(() => false)
  await act(async () => {
    fireEvent.click(queryByTestId('randomizer-randomizeButton'))
    const url = await wait(() => getByTestId('randomizer-restaurantUrl'))
    fireEvent.click(url)
    expect(url).toBeInTheDocument()
  })
```
Not quite sure about this one. Better approach would be to provide initial path to `actRender(<></>, ['initial/path'])` and  use `getPath` to verify that path has not changed.